src/rgb/contract/data.rs

```diff
@@ -290,3 +290,161 @@ pub(super) mod strict_encoding {
+    static U_64: [u8; 9] = [0x3, 0x40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0];
+    static I_8: [u8; 2] = [0x8, 0x8];
+    static I_16: [u8; 3] = [0x9, 0x10, 0x0];
+    static I_32: [u8; 5] = [0xa, 0x20, 0x0, 0x0, 0x0];
+    static I_64: [u8; 9] = [0xb, 0x40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0];
+    static F_32: [u8; 5] = [0x12, 0x14, 0xae, 0x2, 0x42];
+    static F_64: [u8; 9] = [0x13, 0x7b, 0x14, 0xae, 0x47, 0xe1, 0x2a, 0x50, 0x40];
+    static BYTES: [u8; 36] = [
+        0x20, 0x21, 0x0, 0x7c, 0x66, 0xc0, 0x28, 0x3e, 0xe9, 0xbe, 0x98, 0xe, 0x29, 0xce, 0x32,
+        0x5a, 0xf, 0x46, 0x79, 0xef, 0x87, 0x28, 0x8e, 0xd7, 0x3c, 0xe4, 0x7f, 0xc4, 0xf5, 0xc7,
+        0x9d, 0x19, 0xeb, 0xfa, 0x57, 0xda,
+    ];
+    static STRING: [u8; 20] = [
+        0x21, 0x11, 0x0, 0x45, 0x54, 0x48, 0x20, 0x49, 0x53, 0x20, 0x41, 0x20, 0x53, 0x45, 0x43,
+        0x55, 0x52, 0x49, 0x54, 0x59,
+    ];
+    static BITCOIN160: [u8; 21] = [
+        0x48, 0xf, 0x33, 0xe5, 0xdf, 0x8, 0x7c, 0x5c, 0xef, 0x5f, 0xae, 0xbe, 0x76, 0x76, 0xd9,
+        0xe7, 0xa6, 0xb8, 0x2b, 0x4a, 0x99,
+    ];
+    static BITCOIN256: [u8; 33] = [
+        0x49, 0x39, 0x3a, 0x97, 0x1e, 0x46, 0x1d, 0x1c, 0x52, 0xd4, 0xb7, 0xb9, 0xb6, 0x5c, 0x65,
+        0x1a, 0x21, 0x69, 0xf6, 0x82, 0x75, 0x9b, 0x5e, 0xc5, 0xa2, 0x0, 0xde, 0x78, 0x7e, 0x40,
+        0x79, 0x55, 0x7b,
+    ];
+    static SHA256: [u8; 33] = [
+        0x40, 0x99, 0x95, 0x91, 0x96, 0xd2, 0x51, 0x41, 0x94, 0x68, 0x59, 0xbb, 0x21, 0x3e, 0xcc,
+        0x7f, 0x5f, 0xca, 0x55, 0xb8, 0x82, 0x46, 0x7e, 0xb1, 0xd3, 0x9b, 0xf5, 0x88, 0xdf, 0xa8,
+        0x33, 0x2f, 0xa0,
+    ];
+    static SHA512: [u8; 65] = [
+        0x41, 0x67, 0xf7, 0x21, 0x22, 0x5e, 0xfb, 0xd2, 0x5, 0xfc, 0xe, 0x96, 0x70, 0x0, 0x43, 0xc,
+        0x4, 0xa0, 0xe, 0xef, 0x86, 0xa2, 0x9e, 0xdd, 0x40, 0xfa, 0xf4, 0x4e, 0x1b, 0xe2, 0x27,
+        0x75, 0xea, 0xcf, 0x8e, 0x74, 0xfe, 0x87, 0x2f, 0xc0, 0x3d, 0xd4, 0x51, 0x2f, 0x45, 0x15,
+        0xc6, 0xac, 0xa9, 0x7b, 0xb8, 0xf2, 0xf1, 0xf3, 0x84, 0x90, 0xd9, 0x78, 0x6b, 0x4, 0x3e,
+        0x36, 0xed, 0xff, 0x35,
+    ];
+    static PK_BYTES_02: [u8; 34] = [
+        0x81, 0x2, 0x9b, 0x63, 0x47, 0x39, 0x85, 0x5, 0xf5, 0xec, 0x93, 0x82, 0x6d, 0xc6, 0x1c,
+        0x19, 0xf4, 0x7c, 0x66, 0xc0, 0x28, 0x3e, 0xe9, 0xbe, 0x98, 0xe, 0x29, 0xce, 0x32, 0x5a,
+        0xf, 0x46, 0x79, 0xef,
+    ];
+    static SIG_BYTES: [u8; 65] = [
+        0x82, 0xdf, 0x2b, 0x7, 0x1, 0x5f, 0x2e, 0x1, 0x67, 0x74, 0x18, 0x7e, 0xad, 0x4a, 0x4f,
+        0x71, 0x9a, 0x14, 0xe3, 0xe1, 0xad, 0xa1, 0x78, 0xd6, 0x6c, 0xce, 0xcf, 0xa4, 0x5b, 0x63,
+        0x30, 0x70, 0xc2, 0x43, 0xa2, 0xd7, 0x6e, 0xe0, 0x5d, 0x63, 0x49, 0xfe, 0x98, 0x69, 0x6c,
+        0x1c, 0x4d, 0x9a, 0x67, 0x11, 0x24, 0xde, 0x40, 0xc5, 0x31, 0x71, 0xa4, 0xb2, 0x82, 0xb7,
+        0x69, 0xb7, 0xc6, 0x96, 0xcd,
+    ];
+
+    // Test suite function to test against the vectors
+    fn test_suite<T: StrictEncode + StrictDecode + PartialEq + Debug>(
```

Comment:
Maybe we can move this function into `paradigms::strict_encoding` mod and just use it from there? (just make test mode there and this fn `pub(crate)`)

Reply:
Ya could be. I was thinking more like an `util` mod. If there are other such things then they can be combined together in a single `util` for easier reference. I also have few others that can be added too.  Or this can go into `paradigms::strict_encoding` but doesn't feel like the right place for it to me.

Reply:
This function implements encoding functionality, which cant be used anywhere outside of strict encoding-related places, so why strict encoding mod does not feel like the right place for it?

Reply:
Because it's not part of the strict encoding API. I feel ideally library mods should only have code that the user will refer to as API or code for use in other parts of the same module. 
This function is mainly a helper and feels like it will unnecessarily pollute the paradigm mod, as its not part of the API and also its not used for any other part of `paradigm::strict_encoding`. The semantic intent of this function seems more appropriate for a general util module. 

This was my reasoning. But no hard opinion either way.